app/presenters/produce_price_presenter.rb

```diff
@@ -18,12 +19,32 @@ def show
   private
 
   def timeline
-    @produce_prices.all.each do |row| 
+    @items.each do |item|
+      prices = []
+      dates = []
+      units = []
       record = {}
-      record[DATE] = row[DATE]
-      record[row[LABEL]] = row[PRICE].to_f
+
+      @produce_prices.all.where(label: item).each do |row|
```

Comment:
just another note why the sort is useful: because another option to avoid unnecessary db queries would have been to change 

```ruby
@produce_prices.all.where

# to

@produce_prices.to_a # which converts the active record relation to an array and stores it in memory
```
you can then filter on the list and iterate on each filter. this only does one query.

but that would still be an N^2 time complexity because you would be iterating I X P times (where I is the unique item set and P is the produce price result set)

Thats why ordering the set is better because you only iterate on the query results one time and you only incur the cost of one sort. 

and you also dont store the query results in an array you do the order when you do the query so you are utilizing the optimizations that postgres is doing under the hood for querying and sorting your data

hope thats helpful in explaining why the sort/order was worth it.

Reply:
Thanks - this all makes sense. Appreciate the explanations. 

I hadn't thought about the number of times my code would query the db. Good to know and something I will take into consideration going forward. I've made your requested changes, with some minor cosmetic updates (spaces, variable names etc) and pushed in most recent commit. Please review. 

Let me know if anything else needs revising.